src/libraries/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -996,7 +996,7 @@ private static async Task InternalWriteAllTextAsync(StreamWriter sw, string cont
             }
             finally
             {
-                sw.Dispose();
+                sw.Dispose(); // await DisposeAsync
```

Comment:
I don't understand what these comments are trying to say.

Reply:
Ah, I may have figured it out: What the suggested change is that would make the analyzer happier.

So I guess then my comment is "why are they comments instead of code changes?"

Reply:
@bartonjs you figured it out, yes.
My intention with this PR is to show what suggestions this analyzer would give us in runtime, and depending on how noisy it is (meaning how many would be false positives), decide what severity to approve for the [API Proposal](https://github.com/dotnet/runtime/issues/45661#issuecomment-767756782).
Which brings me back to the question:
- Do we need to go through API proposal again or can this be approved offline?
- Based on this level of noise, would it make sense to turn it on in runtime or should we disable it?
- Since it's being noisy, should the severity be info for the rest of the world, or can it be added to roslyn-analyzers as warning?

Reply:
Are they all were false positive? No any valid warnings? Is there any case we could tweak the analyzer to not warn for false positives?

Reply:
> Do we need to go through API proposal again or can this be approved offline?

We don't need to go through API review.

> Based on this level of noise, would it make sense to turn it on in runtime or should we disable it?

We should disable it.

> Since it's being noisy, should the severity be info for the rest of the world, or can it be added to roslyn-analyzers as warning?

It should not be a warning.  The noise we have with this in runtime is due to some common patterns that others use as well, e.g. a single method that has paths for sync and async, use of APIs that have an async alternative but where the developer knows the sync path is sufficient (e.g. FileStream.Dispose for a read-only FileStream), use of Task synchronous waiting methods on tasks known to already be complete, etc.

> No any valid warnings?

It's possible there's a handful, but I doubt more than one or two that are legit, if that.

> Is there any case we could tweak the analyzer to not warn for false positives?

It's possible, but it would need to be imbued with much more smarts.

We should definitely consider splitting functionality into multiple pieces.  For example, I don't believe the warnings that fire for Task.Result/Wait/etc. should be part of this same diagnostic ID.

Reply:
Thanks, @stephentoub !

> We don't need to go through API review.

@terrajobst can you please do the honors and mark [the proposal](https://github.com/dotnet/runtime/issues/45661#issuecomment-767756782) as `api-approved`? (If there are no more concerns, of course).

> We should disable it.
So I'll just close this PR. No changes for runtime.

> It should not be a warning.
I assume this means using the "Info" severity for the rest of the world. Is this correct?

> We should definitely consider splitting functionality into multiple pieces.
No problem, I can start working on that when moving it to roslyn-analyzers.